docs/dev-guides/source-watcher.md

```diff
@@ -0,0 +1,231 @@
+# Watching for source changes
+
+In this guide you'll be developing a Kubernetes controller with
+[Kubebuilder](https://github.com/kubernetes-sigs/kubebuilder)
+that subscribes to [GitRepository](../components/source/gitrepositories.md)
+events and reacts to revision changes by downloading the artifact produced by 
+[source-controller](../components/source/controller.md).
+
+## Prerequisites
+
+On your dev machine install the following tools:
+
+* go >= 1.13
+* kubebuilder >= 2.3
+* kind >= 0.8
+* kubectl >= 1.18
+* kustomize >= 3.5
+* docker >= 19.03
+
+## Install the GitOps toolkit
+
+Create a cluster for testing:
+
+```sh
+kind create cluster --name dev
+```
+
+Install the toolkit CLI:
+
+```sh
+curl -s https://toolkit.fluxcd.io/install.sh | sudo bash
+```
+
+Verify that your dev machine satisfies the prerequisites with:
+
+```sh
+tk check --pre
+```
+
+Install the toolkit controllers on the dev cluster:
+
+```sh
+tk install
+```
+
+## Clone the sample controller
+
+You'll be using [stefanprodan/source-watcher](https://github.com/stefanprodan/source-watcher) as
+a template for developing your own controller. The source-watcher was scaffolded with `kubebuilder init`.
+
+Clone the source-watcher repo:
+
+```sh
+git clone https://github.com/stefanprodan/source-watcher
+cd source-watcher
+```
+
+Build the controller:
+
+```sh
+make
+```
+
+## Run the controller
+
+Port forward to source-controller artifacts server:
+
+```sh
+kubectl -n gitops-system port-forward svc/source-controller 8080:80
+```
+
+Export the local address as `SOURCE_HOST`:
+
+```sh
+export SOURCE_HOST=localhost:8080
```

Comment:
port 8080 clashes with the source-watcher's metrics address listener, I used 9090 and it worked fine

Reply:
Good catch!